File: app/src/main/java/com/permissionnanny/ConfirmRequestView.java
package com.permissionnanny;

import android.app.Activity;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.ViewStub;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;
import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnCheckedChanged;
import butterknife.OnClick;

/**
 *
 */
public class ConfirmRequestView {
    private final Activity mActivity;
    private final TextDialogStubView mTextStub;

    private ConfirmRequestBinder mBinder;

    @Bind(R.id.tvTitle) TextView tvTitle;
    @Bind(R.id.ivIcon) ImageView ivIcon;
    @Bind(R.id.vsStub) ViewStub vsStub;
    @Bind(R.id.remember) CheckBox cbRemember;
    @Bind(R.id.btnPositive) Button btnAllow;
    @Bind(R.id.btnNegative) Button btnDeny;

    public ConfirmRequestView(Activity activity, ConfirmRequestBinder binder, TextDialogStubView stub) {
        mActivity = activity;
        mBinder = binder;
        mTextStub = stub;
    }

    public void preOnCreate(Bundle state) {
        mActivity.setTheme(R.style.Nanny_Light_Dialog_NoActionBar);
    }

    public void onCreate(Bundle state) {
        mActivity.setContentView(R.layout.dialog);
        ButterKnife.bind(this, mActivity);
        mTextStub.inflateViewStub(vsStub);
        if (Build.VERSION.SDK_INT >= 11) {
            mActivity.setFinishOnTouchOutside(false);
        }
        cbRemember.setText(R.string.dialog_remember);
    }

    public void bindViews() {
        tvTitle.setText(mBinder.getDialogTitle());
        Drawable icon = mBinder.getDialogIcon();
        ivIcon.setVisibility(icon == null ? View.GONE : View.VISIBLE);
        ivIcon.setImageDrawable(icon);
        boolean remember = mBinder.getRememberPreference();
        btnAllow.setText(remember ? R.string.dialog_always_allow : R.string.dialog_allow);
        btnDeny.setText(remember ? R.string.dialog_always_deny : R.string.dialog_deny);
        mTextStub.bindViews();
    }

    @OnCheckedChanged(R.id.remember)
    void onRememberPreferenceChanged(boolean isChecked) {
        mBinder.changeRememberPreference(isChecked);
    }

    @OnClick(R.id.btnPositive)
    void onAllow() {
        mBinder.executeAllow();
        mActivity.finish();
    }

    @OnClick(R.id.btnNegative)
    void onDeny() {
        mBinder.executeDeny();
        mActivity.finish();
    }
}

File: app/src/main/kotlin/com/permissionnanny/ConfirmRequestView.kt
package com.permissionnanny

import android.app.Activity
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.ViewStub
import android.widget.Button
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import org.jetbrains.anko.find

/**

 */
class ConfirmRequestView(
        private val activity: Activity,
        private val binder: ConfirmRequestBinder,
        private val textStub: TextDialogStubView) {

    lateinit var tvTitle: TextView
    lateinit var ivIcon: ImageView
    lateinit var vsStub: ViewStub
    lateinit var cbRemember: CheckBox
    lateinit var btnAllow: Button
    lateinit var btnDeny: Button

    fun preOnCreate(state: Bundle?) {
        activity.setTheme(R.style.Nanny_Light_Dialog_NoActionBar)
    }

    fun onCreate(state: Bundle?) {
        activity.setContentView(R.layout.dialog)
        tvTitle = activity.find(R.id.tvTitle)
        ivIcon = activity.find(R.id.ivIcon)
        vsStub = activity.find(R.id.vsStub)
        cbRemember = activity.find(R.id.remember)
        btnAllow = activity.find(R.id.btnPositive)
        btnDeny = activity.find(R.id.btnNegative)
        textStub.inflateViewStub(vsStub)
        if (Build.VERSION.SDK_INT >= 11) {
            activity.setFinishOnTouchOutside(false)
        }
        cbRemember.setText(R.string.dialog_remember)

        cbRemember.setOnCheckedChangeListener { view, isChecked -> onRememberPreferenceChanged(isChecked) }
        btnAllow.setOnClickListener { onAllow() }
        btnDeny.setOnClickListener { onDeny() }
    }

    fun bindViews() {
        tvTitle.text = binder.dialogTitle
        val icon = binder.dialogIcon
        ivIcon.visibility = if (icon == null) View.GONE else View.VISIBLE
        ivIcon.setImageDrawable(icon)
        val remember = binder.rememberPreference
        btnAllow.setText(if (remember) R.string.dialog_always_allow else R.string.dialog_allow)
        btnDeny.setText(if (remember) R.string.dialog_always_deny else R.string.dialog_deny)
        textStub.bindViews()
    }

    internal fun onRememberPreferenceChanged(isChecked: Boolean) {
        binder.changeRememberPreference(isChecked)
    }

    internal fun onAllow() {
        binder.executeAllow()
        activity.finish()
    }

    internal fun onDeny() {
        binder.executeDeny()
        activity.finish()
    }
}
